aptos-move/framework/src/natives/aggregator_natives/context.rs

```diff
@@ -14,6 +14,8 @@ use std::{
     collections::{btree_map, BTreeMap},
 };
 
+pub type TxnIndex = u32;
```

Comment:
I think we need to be clear on what this is. 
This isn't a txn_index from the start of the chain, this isn't a version txn will get.
This isn't even a txn_index within a block, as state sync might not respect block boundaries.

this is transaction index within a batch that BlockSTM executes simultaneously. Do we have a name for that @gelash ? 

We should at least put a long comment here, and if there is a name for that maybe we should even rename the type for TxnIndexInExecutionBatch or something.

Reply:
Inside BlockExecutor TxnIndex is defined as u32, but that's for the purposes of the block executor.
It is indeed just an index within the block as given to the block executor, it doesn't have a deep meaning when it arrives (i.e. right before getting there some txns may get discarded due to signature checks), and doesn't have meaning after the block is executed (as again I believe some txns can get discarded based on output). It's just some unique number within block execution.

I do have some concern now for passing the txn indices to the session, as it can be easy for other developers to take it and use it for something, and run into terrible determinism issues. Plus for simulation it is set to 0, a lot of other places too. Seems troublesome, @zekun000 did you have such an implementation in mind? Wouldn't it be cleaner to get the txn index within session via a trait call to the view and then generate, or just get the id directly and then have a way to generate subsequent IDs? Plus it aligns with how we grab ids for replacement from storage.

Reply:
Yeah. I'm not a fan of passing 0 as `txn_idx` in many places as well. Earlier, instead of passing 0, I tried to pass actual `txn_idx`. But then there were many places where it's not clear what txn_idx to pass. So, when I was facing trouble, @zekun000 suggested to pass `txn_idx` as 0.
https://aptos-org.slack.com/archives/C057VPCD7QX/p1689114447788319?thread_ts=1689054263.243439&cid=C057VPCD7QX

Reply:
@gelash Can you please explain how to generate txn_idx with session via trait call to view? What does `view` mean here?

Reply:
I must say I am also not a fan of passing 0 everywhere (or at least explicitly, maybe hiding behind enum at least would be better). A general (possibly stupid) question for my understanding of IDs: why do we not want to have a single generator somewhere which we can query to get unique ids?